### PR TITLE
Build correct grading service URLs in Moodle

### PR DESCRIPTION
**Pull request by marcospri**
For https://github.com/hypothesis/lms/issues/3909

A small fix to correctly build URLs, just appending `/something` won't work in Moodle.

### CODE REVIEW

File: lms/services/lti_grading/_v13.py [hunk 1]
```diff
@@ -54,5 +55,20 @@ def record_result(self, grading_id, score=None):  # pylint:disable=arguments-dif
                 "activityProgress": "Completed",
                 "gradingProgress": "FullyGraded",
             },
-            headers={"Content-Type": "application/vnd.ims.lis.v2.lineitem+json"},
+            headers={"Content-Type": "application/vnd.ims.lis.v1.score+json"},
```
Comment (marcospri) [inline lms/services/lti_grading/_v13.py:58]: This values was wrong: https://www.imsglobal.org/spec/lti-ags/v2p0#example-posting-a-final-score-update

I guess both blackboard and IMS didn't care about that. Moodle does. I retested in the other ones and still works with the right value.

File: tests/unit/lms/services/lti_grading/_v13_test.py [hunk 2]
```diff
@@ -94,4 +118,4 @@ def response(self):
 
     @pytest.fixture
     def svc(self, ltia_http_service):
-        return LTI13GradingService(self.SERVICE_URL, ltia_http_service)
+        return LTI13GradingService("http://example.com/service_url", ltia_http_service)
```
Comment (marcospri) [inline tests/unit/lms/services/lti_grading/_v13_test.py:121]: Can't be a `sentinel`. Needs to be a string. But we'd only use `SERVICE_URL` here

File: lms/services/lti_grading/_v13.py [hunk 3]
```diff
@@ -22,7 +23,7 @@ def read_result(self, grading_id):
         try:
             response = self._ltia_service.request(
                 "GET",
-                self.grading_url + "/results",
+                self._service_url(self.grading_url, "/results"),
                 scopes=self.LTIA_SCOPES,
                 params={"user_id": grading_id},
```
Comment (jon-betts) [inline lms/services/lti_grading/_v13.py:28]: Note to self: I've given this a quick check and requests will take care of merging the params here into those on the provided URL

File: tests/unit/lms/services/lti_grading/_v13_test.py [hunk 4]
```diff
@@ -8,16 +8,28 @@
...
+    @pytest.mark.parametrize(
+        "service_url,results_url",
+        [
+            ("https://lms.com/lineitems", "https://lms.com/lineitems/results"),
+            (
+                "https://lms.com/lineitems?param=1",
+                "https://lms.com/lineitems/results?param=1",
+            ),
+        ],
+    )
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v13_test.py:21]: I think you could just use the more difficult example here, as it's a superset of everything going on in the other example.

File: tests/unit/lms/services/lti_grading/_v13_test.py [hunk 5]
```diff
@@ -60,12 +72,24 @@ def test_read_bad_response_lti_result(self, svc, ltia_http_service, bad_response
...
+    @pytest.mark.parametrize(
+        "service_url,scores_url",
+        [
+            ("https://lms.com/lineitems", "https://lms.com/lineitems/scores"),
+            (
+                "https://lms.com/lineitems?param=1",
+                "https://lms.com/lineitems/scores?param=1",
+            ),
+        ],
+    )
```
Comment (jon-betts) [inline tests/unit/lms/services/lti_grading/_v13_test.py:84]: Same here. The second case covers the first as well.

**Review by jon-betts — APPROVED** [review]
Yep this looks good.

I think the tests could be slimmed down to the example with the param instead of parameterising, as it covers everything about the previous case as well.

**Pull request merged**